### PR TITLE
Start of addressing SB #115, #124, #123

### PR DESCRIPTION
**Pull request by mvahowe**
@rdb This is currently untested and probably needs reformatting too. But could you take a look at

```metadata.schema.json:9```

? I'm not sure how to specify an object with 1 or more key, the value of each key being another object.

### CODE REVIEW

**Comment by jag3773** [comment]
@mvahowe Please rebase this to merge into `develop`.  I think I should change our default branch to `develop` so that this is an easier process...

**Comment by mvahowe** [comment]
@jag3773 There's no develop at this address :-) I'll make a PR on SB once I have something that works here.

**Comment by rdb** [comment]
@mvahowe it'd help me if you showed an example of the syntax you were going for here.

**Comment by mvahowe** [comment]
Good point, I updated the textTranslation test document.

**Comment by rdb** [comment]
@mvahowe additionalProperties defines a schema for the keys, not the values.  You just need something like this instead:
```json
        "properties": {
            "id": {"$ref": "common.schema.json#/definitions/url"},
            "name": {"$ref": "common.schema.json#/definitions/localizedText"}
        },
```

**Comment by rdb** [comment]
@mvahowe Please disregard my previous comment.  Based on the example, you can just put an object schema under additionalProperties.  Something like this:
```json
        "idServers": {
            "type": "object",
            "additionalProperties": {
                "type": "object",
                "properties": {
                    "id": {"$ref": "common.schema.json#/definitions/url"},
                    "name": {"$ref": "common.schema.json#/definitions/localizedText"}
                },
                "required": ["id"],
                "additionalProperties": false
            },
            "minProperties": 1
        }
```

**Comment by mvahowe** [comment]
@rdb Thanks. The latest works, but I think the idServers part of the schema is underspecified (unexpected keys, at least one key, format of keys...)

File: code/fs_metadata_store.js [hunk 1]
```diff
@@ -23,12 +24,22 @@ class FSMetadataStore extends MetadataStore {
      */
     __idServersDetails() {
 	const ret = {};
-	Object.keys(this._urls).forEach(function (ids) {
-	    ret[ids] = {"id": ids};
-	});
+	const urlKeys = Object.keys(this._urls);
+	const nKeys = urlKeys.length;
+	for (var n=0; n < nKeys; n++) {
```
Comment (rdb) [inline code/fs_metadata_store.js]: FYI, this is cleaner: `for (const key of urlKeys) {`

**Comment by rdb** [comment]
You can use propertyNames to define a schema for the keys, and minProperties to specify a minimum number of keys.

**Comment by mvahowe** [comment]
@rdb Thanks. I'll merge this and make a PR against the schema repo.

**Pull request merged**